snuba/datasets/__init__.py

```diff
@@ -93,6 +93,21 @@ def get_prewhere_keys(self) -> Sequence[str]:
         """
         return []
 
+    def get_min_columns(self) -> Sequence[str]:
+        """
+        Return a list of the smallest subset of columns for a dataset.
+
+        This subset must include an event_id and timestamp type column, and
+        is used by the query splitter to help optimize loading wide results.
+        """
+        return NotImplementedError('dataset does not support get_min_columns')
+
+    def get_timestamp_column(self) -> str:
```

Comment:
I think this is going to create a problem with datasets that are not time series (@manuzope, for your PR).
If a dataset is not a time series we should not apply time series split at all, and there would be no meaningful value to return from this method.

I would have made it a little more abstract, letting the dataset parent class return a representation of which split policies to run and which parameters to use.

Also I guess this breaks the groups dataset that was committed shortly before this PR, since it does not provide an implementation of this method.
Also when a dataset returns a table name, it should also be able to deal with a table alias. See above on column_expr.
All this stuff is new from 30 minutes before this PR was committed, but we should fix it.

Reply:
Do you see the query splitting strategies moving into the dataset long term? The current col_split approach is closely coupled to data that can be ordered by time so that the `to_date` and `from_date` conditions can be clamped based on minimal column set query.
If we moved all the splitting logic into the dataset some of those problems could go away.

Reply:
Short term:
- yes, the dataset needs to drive which split policies we apply simply because there are datasets where time split makes no sense. Whether the dataset runs the logic itself or provides the parameter is a separate concern.
Long term:
- We are introducing some more structured query processing/optimization, split should be there, but it would still be directed by the dataset. But this would happen much later.